lambda_src/drivers/destination_s3.py

```diff
@@ -31,7 +32,13 @@ def parse_destination_uri(destination: Text) -> Tuple[Text, Text]:
     LOG.debug(f'destination from header is {destination}')
     parsed_url = urlparse(destination)
     bucket = parsed_url.netloc
-    prefix = parsed_url.path.split('/', 2)[1] if parsed_url.path.count('/') >= 2 else ''
+        
+    parts = parsed_url.path.split('/')
+    prefix_parts = [part for part in parts if not part.startswith('%') and part]
+    strftime_parts = [time.strftime(part) for part in parts if part.startswith('%')]
+
+    prefix_with_time = "/".join(prefix_parts + strftime_parts)
+    prefix = prefix_with_time if parsed_url.path.count('/') >= 1 else ''
```

Comment:
`s3://dev-bucket/geff-function-3-results` 

Considering this to be the destination URI, it wouldn't be actually, since `urlparse` returns `/geff-function-3-results` as the path, which has a single `/`.

Reply:
what is the intent of this code, then? is it to handle the case where the user provides a destination that's just a simple string like `"dev-bucket"`?

Reply:
default prefix for data should be `/data` and not empty. So that we don't end up with a lot of files in the bucket home dir in the UI.

Reply:
in the case of a destination prefix, having a default seems to be non-obvious. if someone says `s3://foo/` do we add the default or did they explicitly want the root by ending it with a `/`? is that something we should document, that there's a difference between this and `s3://foo`, which really means `s3://foo/data`?

I think the easiest thing if we have a `destination-uri` is just to respect what the user requests -- put the data there in whatever format makes sense, with an optional translation layer via parameters and strftime flags, etc.